### PR TITLE
add metric for dispatch duration in nats dispatcher

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- introduce a builtin (not on istio depending) metric for dispatch duration in eventin controller 

**Related issue(s)**

### CODE REVIEW

**Review by raypinto — COMMENTED** [review]
We also need to document this metric!

File: components/eventing-controller/pkg/backend/metrics/collector.go [hunk 1]
```diff
@@ -2,12 +2,17 @@ package metrics
...
 	"fmt"
+	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"sigs.k8s.io/controller-runtime/pkg/metrics"
 )
 
 const (
+	// latencyMetricKey name of the dispatch_duration metric
+	latencyMetricKey = "eventing_ec_subscriber_dispatch_duration_milliseconds"
```
[the remark this answers is not in the document]
Comment (k15r) [reply components/eventing-controller/pkg/backend/metrics/collector.go:13]: then i suggest using:
`eventing_ec_nats_subscriber_dispatch_duration_milliseconds`

we should migrate all metrics to follow a uniform pattern

`eventing_<component>_<descriptive part for the metric>`

if we start with `nats` then our names might be confused as metrics exported by nats server

[the remark this answers is not in the document]
Comment (raypinto) [reply components/eventing-controller/pkg/backend/metrics/collector.go:13]: I agree! lets do that

**Review by raypinto — COMMENTED** [review]
(nit): I don't see the metric `eventing_ec_nats_subscriber_dispatch_duration_seconds` in the FI tests. What do you think about adding it there?

**Comment by k15r** [comment]
> (nit): I don't see the metric `eventing_ec_nats_subscriber_dispatch_duration_seconds` in the FI tests. What do you think about adding it there?

its currently not used in any dashboard. as soon as we use it there, we can think about such tests

**Review by raypinto — APPROVED** [review]

File: docs/04-operation-guides/operations/evnt-02-eventing-metrics.md [hunk 2]
```diff
@@ -7,23 +7,23 @@ The metrics follow the [Prometheus naming convention](https://prometheus.io/docs
...
-|---------------------------------------------|:-------------------------------------------------------------------------------|
-| **nats_ec_event_type_subscribed_total**     | Total number of all the eventTypes subscribed using the Subscription CRD.      |
-| **nats_ec_delivery_per_subscription_total** | Total number of dispatched events per subscription, with status code and sink. |
+| Metric                                                    | Description                                                                                                                 |
+| --------------------------------------------------------- | :-------------------------------------------------------------------------------------------------------------------------- |
+| **eventing_ec_nats_delivery_per_subscription_total**      | The total number of dispatched events per subscription                                                                      |
+| **eventing_ec_nats_subscriber_dispatch_duration_seconds** | The duration of sending an incoming nats message to the subscriber (not including processing the message in the dispatcher) |
```
Comment (NHingerl) [inline docs/04-operation-guides/operations/evnt-02-eventing-metrics.md:24]:
```suggestion
| **eventing_ec_nats_subscriber_dispatch_duration_seconds** | The duration of sending an incoming NATS message to the subscriber (not including processing the message in the dispatcher) |
```

**Review by mfaizanse — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Pull request merged**